app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        __class__.people[self.name] = self
+
+
+def create_person_list(peoples: list) -> list:
+    list_result = [Person(member["name"], member["age"]) for member in peoples]
+
+    for index, member in enumerate(peoples):
+        if "wife" in member:
+            for pos in range(len(peoples)):
+                if list_result[pos].name == member["wife"]:
+                    list_result[index].wife = list_result[pos]
+        if "husband" in member:
+            for pos in range(len(peoples)):
+                if list_result[pos].name == member["husband"]:
+                    list_result[index].husband = list_result[pos]
```

Comment:
Use one loop and do not apply index in your solution, retrieve value by key where needed from `Person.people`. Besides that, you could incorporate `.get` method and walrus operator into your solution